packages/integration_test/android/src/main/java/dev/flutter/plugins/integration_test/FlutterDeviceScreenshot.java

```diff
@@ -51,8 +55,15 @@ class FlutterDeviceScreenshot {
    * @return the Flutter view.
    */
   @Nullable
-  private static FlutterView getFlutterView(@NonNull Activity activity) {
-   return (FlutterView)activity.findViewById(FlutterActivity.FLUTTER_VIEW_ID);
+  @VisibleForTesting
+  public static FlutterView getFlutterView(@NonNull Activity activity) {
+    if (activity instanceof FlutterActivity) {
```

Comment:
Should this be ...
```suggestion
    if (activity instanceof FlutterActivity ||| activity instanceOf FlutterFragmentActivity) {
```
Is it possible to avoid the instance of check at all? 
go/UnhealthyIdioms#awkward-inheritance-hierarchy

Reply:
> Should this be ...

No, note that the cases are indeed different here:
`findViewById(FlutterActivity.FLUTTER_VIEW_ID)` vs `findViewById(FlutterFragment.FLUTTER_VIEW_ID)` (notably, Flutter _Activity_ vs Flutter _Fragment_

Reply:
I am not aware of a better way to do this without `instanceof` (I try to avoid it in general). Open to ideas if I'm missing something though.

We could just do something like (pseudo code)
```
if (<findbyviewid isn't null for FlutterActivity.FLUTTER_VIEW_ID>)
<return that>
if (<findbyviewid isn't null for FlutterFragmentFLUTTER_VIEW_ID>)
<return that>
<return null>
```
But if for whatever reason we are actually dealing with a FlutterFragmentActivity, and there is a view registered with the `FlutterActivity.FLUTTER_VIEW_ID`, then we would return the wrong view.

Reply:
After some further consideration if you want to move on feel free to file a follow-up ticket and merge this as is.

Reply:
https://github.com/flutter/flutter/issues/134252